packages/protocol/contracts/governance/Governance.sol

```diff
@@ -510,13 +512,14 @@ contract Governance is IGovernance, Ownable, Initializable, UsingBondedDeposits,
     dequeueProposalsIfReady();
     Proposal storage proposal = proposals[proposalId];
     require(_proposalExists(proposal) && dequeued[index] == proposalId);
-    if (isDequeuedProposalExpired(proposalId)) {
-      deleteDequeuedProposal(proposalId, index);
+    if (isDequeuedProposalExpired(proposal)) {
+      deleteDequeuedProposal(proposal, proposalId, index);
       return false;
     }
-    ProposalStage stage = _getDequeuedProposalStage(proposal.timestamp);
+    ProposalStage stage = getDequeuedProposalStage(proposal.timestamp);
     require(msg.sender == approver && !proposal.approved && stage == ProposalStage.Approval);
     proposal.approved = true;
+    proposal.totalWeight = totalWeight();
```

Comment:
What's the issue with just snapshotting the `totalWeight` upon approval, and *only* upon approval?

Reply:
That does simplify voting, but we run some risk that the recorded `totalWeight` becomes very inconsistent with the votes cast on the proposal, particularly since Referendum is relatively long. It's theoretically possible that by bonding and voting "yes" with gold post-approval, the participation is inflated because we're using the `totalWeight` snapshotted at approval, so the proposal is easier to pass. This might also produce strange participation baseline updates. Whether or not the numbers make this sufficiently problematic is another question.